lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/SQLUtils.java

```diff
@@ -1434,9 +1442,24 @@ public static String createSelectInsertFromObjectReplication(Long viewId, List<C
 			builder.append(" AND ").append(OBJECT_REPLICATION_ALIAS).append(".").append(OBJECT_REPLICATION_COL_OBJECT_ID)
 					.append(" IN (:").append(ID_PARAM_NAME).append(")");
 		}
-		builder.append(" GROUP BY ").append(OBJECT_REPLICATION_ALIAS).append(".").append(OBJECT_REPLICATION_COL_OBJECT_ID);
-		builder.append(" ORDER BY ").append(OBJECT_REPLICATION_ALIAS).append(".").append(OBJECT_REPLICATION_COL_OBJECT_ID);
-		return headers;
+	}
+
+	/**
+	 * Generate the SQL to validate that all of the list columns for a view table from the object replication tables.
+	 * @param viewId
+	 * @param viewTypeMask
+	 * @param annotationNames
+	 * @return
+	 */
+	public static String createAnnotationMaxListLengthSQL(Long viewId, Long viewTypeMask,
+																 Set<String> annotationNames, boolean filterByRows) {
+		ValidateArgument.requiredNotEmpty(annotationNames,"annotationNames");
+
+		StringBuilder builder = new StringBuilder();
+		builder.append("SELECT ");
+		buildSelectListLength(builder, annotationNames);

```

Comment:
We might want to actually do a group by A.ANNO_KEY with the count, so that we can simplify the select statement.